components/molecules/Modal/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FunctionComponent } from 'react';
+import ReactModal from 'react-modal';
+import styled from '@emotion/styled';
+
+import Button from 'components/atoms/Button/index';
+
+type ModalProps = {
+    children: React.ReactNode;
+    isOpen: boolean;
+    contentType: string;
+    onClose: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+const Modal: FunctionComponent<ModalProps> = (props) => {
+    const { children, isOpen, contentType, onClose } = props;
+    return (
+        <ModalWrapper isOpen={isOpen} onRequestClose={onClose} closeTimeoutMS={2000}>
+            <Button onClick={onClose}>닫기</Button>
+            <Content className={contentType}>{children}</Content>
+        </ModalWrapper>
+    );
+};
+
+const ModalWrapper = styled(ReactModal)`
+    position: absolute;
+    display: flex;
+    flex-direction: column;
+    background-color: ${({ theme }) => theme.buttonColors.primary};
+    color: ${({ theme }) => theme.colors.primary};
+    margin: 80px 460px;
+    width: 1000px;
+    min-height: 80px;
+    max-height: 878px;
+    outline: none;
+    transition: all 0.3s ease-in-out;
+    &[class*='after-open'] {
+        opacity: 1;
+    }
+    &[class*='before-close'] {
+        opacity: 0;
+    }
+    > Button {
+        margin: 16px 20px 22px auto;
+    }
+`;
+
+const Content = styled.div`
+    display: flex;
+    justify-contents: center;
+    overflow: auto;
+    margin-bottom: 1rem;
+    padding: ${(props) => (props.className === 'search' ? '0 40px' : '0 56px')};

```

Comment:
```suggestion
    padding: 0 56px;
    &.search {
        padding: 0 40px;
    }
```

class name으로 하자면 이렇게 할수 있을꺼 같고
따로 props를 줘서 하는 방식도 좋을것 같습니다.

Reply:
아 넵 감사합니다! 숙지해놓겠습니당.